cli/src/cmd/forge/verify.rs

```diff
@@ -262,88 +423,39 @@ impl VerifyCheckArgs {
         let etherscan = Client::new(self.chain.try_into()?, &self.etherscan_key)
             .wrap_err("Failed to create etherscan client")?;
 
-        let resp = etherscan
-            .check_contract_verification_status(self.guid.clone())
+        println!("Waiting for verification result...");
+        let retry = self.retry.clone();
+        retry
+            .run_async(|| {
```

Comment:
Maybe for a followup PR, but maybe we should pull this logic into ethers-etherscan